lib/mailjet/mailer.rb

```diff
@@ -27,25 +27,187 @@ def initialize(options = {})
+    # The mailjet related ones are stocked in another variable and directly added to the body
+    if mail.header && !mail.header.fields.empty?
+      content[:Headers] = {}
+      mail.header.fields.each do |header|
+        if header.name.start_with?('X-') && !header.name.start_with?('X-MJ') && !header.name.start_with?('X-Mailjet')
+          content[:Headers][header.name] = header.value
+        end
+      end
+    end
+
+    # Reply-To is not a property in Mailjet Send API
+    # Passing it as an header if mail.reply_to
+    
+    if mail.reply_to
+      if mail.reply_to.display_names.first
+        content[:Headers]['Reply-To'] = {:Email=> mail[:reply_to].addresses.first, :Name=> mail[:reply_to].display_names.first}
+      else
+        content[:Headers]['Reply-To'] = {:Email=> mail[:reply_to].addresses.first}
+      end
+    end
+    
+    if (mail[:to])
+      if (mail[:to].is_a? String)
+        if mail[:to].display_names.first
+          to = [{:Email=>mail[:to].addresses.first, :Name=>mail[:to].display_names.first}]
+        else
+          to = [{:Email=>mail[:to].addresses.first}]
+        end
+      else
+        to = []
+        mail[:to].each do |t|
+          if (t.display_name)
+            to << {:Email=> t.address, :Name=>t.display_name}
+          else
+            to << {:Email=> t.address}
+          end
+        end
+      end
+    end
+    
+    base_from = {
+      From:{
+        :Email=> mail[:from].addresses.first
+      }
+    }
+    if (mail[:from].display_names.first)
+      base_from[:From][:Name] = mail[:from].display_names.first
+    end
+  
+    if (mail[:cc])
+      if (mail[:cc].is_a? String)
+        if mail[:cc].display_name.first
+          ccs =[{:Email=>mail[:cc].address.first, :Name=>mail[:cc].display_name.first}]
+        else
+          ccs =[{:Email=>mail[:cc].address.first}]
+        end
+      else
+        ccs = []
+        mail[:cc].each do |cc|
+          ccs << {:Email=> cc.address, :Name=>cc.display_name}
+        end
+      end
+    end
+    
+    if (mail[:bcc])
+      if (mail[:bcc].formatted.is_a? String)
+        if mail[:bcc].display_name.first
+          payload[:Bcc] = [{:Email=>mail[:bcc].address.first, :Name=>mail[:bcc].display_name.first}]
+        else
+          payload[:Bcc] = [{:Email=>mail[:bcc].address.first}]
+        end
+      else
+        bccs = []
+        mail[:bcc].formatted.each do |bcc|
+          if bcc.display_name
+            bccs << {:Email=> bcc.address, :Name=>bcc.display_name}
+          else
+            bccs << {:Email=> bcc.address}
+          end
+        end
+      end
+    end
+    
+    payload = {
+      :To=> to,
+      :Sender=> mail[:sender],
+      :Subject=> mail.subject
+    }.merge(content)
+    .merge(base_from)
+    .merge(@delivery_method_options_v3_1)
+    
+    if (mail[:cc])
+      payload[:Cc] = ccs
+    end
+  
+    if (mail[:bcc])
+      payload[:Bcc] = bccs
+    end
```

Comment:
you could use here the inline syntax for `if`:
```ruby
payload[:Cc] = ccs if mail[:cc]
payload[:Bcc] = bccs if mail[:bcc]
```
but it could also be set when you handle both of these cases above